modules/basicfuncs/numeric-funcs.c

```diff
@@ -130,3 +132,156 @@ tf_num_mod(LogMessage *msg, gint argc, GString *argv[], GString *result)
 }
 
 TEMPLATE_FUNCTION_SIMPLE(tf_num_mod);
+
+
+static GString *
+_make_gstring_buffer(const LogTemplateInvokeArgs *args)
+{
+  if (args->bufs->len == 0)
+    g_ptr_array_add(args->bufs, g_string_sized_new(64));
+
+  return (GString *) g_ptr_array_index(args->bufs, 0);
+}
+
```

Comment:
I moved this function into `template/function.h`. It's not the best place, because LogTemplateInvokeArgs is initialized in `log_template_append_format_with_context` without "taking ownership" of its `GPtrArray *bufs` attribute. Do you have any better idea?